plugins/dell-dock/fu-dell-dock-plugin.c

```diff
@@ -51,101 +57,104 @@ struct _FuDellDockPlugin {
 
 G_DEFINE_TYPE(FuDellDockPlugin, fu_dell_dock_plugin, FU_TYPE_PLUGIN)
 
+static FuDellDockBaseType
+fu_dell_dock_plugin_get_dock_type(FuDevice *device, GError **error)
+{
+	if (FU_IS_DELL_DOCK_EC(device))
+		return fu_dell_dock_get_dock_type(device);
+
+	return FU_DELL_DOCK_BASE_TYPE_UNKNOWN;
+}
+
 static gboolean
-fu_dell_dock_plugin_create_node(FuPlugin *plugin, FuDevice *device, GError **error)
+fu_dell_dock_plugin_probe_ec_v1_mst(FuPlugin *plugin,
+				    FuDevice *ec_device,
+				    guint8 dock_type,
+				    GError **error)
 {
+	FuContext *ctx = fu_plugin_get_context(plugin);
+	const gchar *instance_id;
+	g_autoptr(FuDellDockMst) mst_device = fu_dell_dock_mst_new(ctx);
 	g_autoptr(FuDeviceLocker) locker = NULL;
 
-	locker = fu_device_locker_new(device, error);
+	instance_id = fu_dell_dock_get_instance_id(dock_type, dock_component_mst, 0, 0);
+	fu_device_add_instance_id(FU_DEVICE(mst_device), instance_id);
+	locker = fu_device_locker_new(FU_DEVICE(mst_device), error);
```

Comment:
after some tests, the `mst` device in the `setup` lifecycle time looks for its `parent` and if `parent` is unset at that time will end up in a segmentation fault.